### PR TITLE
Support for `rjsf form` in sistent modal and migrate all modals

### PR DESCRIPTION
**Pull request by sudhanshutech**
**Notes for Reviewers**

This PR helps to use rjsf wrapper inside sistentmodal
image

### Other migration of modals
- [x] Connections in environment
- [x] Environmemt in workspaces
- [x] designs in workspaces
- [x] Edit and create of workspace
- [x] edit and delete of environment
- [x] Import and publish of design
- [x] import and publish of filters
- [x] Performance profile modals

image
image

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by dragon-slayer875** [comment]
Tested for transferList components. Works. :+1:

**Comment by Deepak0320Singhal** [comment]
@sudhanshutech Thank you for your contributions.
Let's discuss this on Websites's call. Adding this as an agenda item into the [meeting minutes](https://docs.google.com/document/d/1XczAHXVe2FIWPqiF57ospJ43zw5cZQ7ui8mn39v5EvA/edit).

**Comment by leecalcote** [comment]
> Tested for transferList components. Works. 👍

Thank you, @dragon-slayer875

**Comment by leecalcote** [comment]
@captain-Akshay does this PR have your yay or neigh?

**Comment by dragon-slayer875** [comment]
> @captain-Akshay does this PR have your yay or neigh?

Sudhanshu is going to push some changes in this PR. After that it can be tested.

**Comment by sudhanshutech** [comment]
@dragon-slayer875 its ready ✅

File: ui/components/Modal.js [hunk 1]
```diff
@@ -290,3 +291,110 @@ function Modal(props) {
...
+
+  return (
+    <>
+      <ModalBody>
+        {loadingSchema ? (
+          <div style={{ textAlign: 'center', padding: '8rem 17rem' }}>
+            <CircularProgress />
+          </div>
+        ) : (
+          <RJSFWrapper
```
Comment (sudhanshutech) [inline ui/components/Modal.js:356]: @aabidsofi19 @Yashsharma1911  here what i am saying that just wrapping the existing rjsfwrapper with sistent modal

File: ui/components/Modal.js [hunk 2]
```diff
@@ -290,3 +291,110 @@ function Modal(props) {
...
+    setCanNotSubmit(false);
+    const handleDesignNameCheck = () => {
+      const designName = title?.toLowerCase();
+      const forbiddenWords = ['untitled design', 'Untitled', 'lfx'];
+
+      for (const word of forbiddenWords) {
+        if (designName?.includes(word)) {
+          setSnackbar({
+            severity: 'warning',
+            message: `Design name should not contain Untitled Design, Untitled, LFX`,
```
Comment (aabidsofi19) [inline ui/components/Modal.js:322]: whats the Untitled LFX ?

File: ui/components/Modal.js (hunk 1 above)
Comment (aabidsofi19) [reply ui/components/Modal.js:356]: extract the rjsf part only .  eg how will you handle the case for secondary and tertiary buttons ?

**Review by aabidsofi19 — CHANGES_REQUESTED** [review]

File: ui/components/Modal.js (hunk 2 above)
Comment (sudhanshutech) [reply ui/components/Modal.js:322]: i am not sure about this, this was added by @Yashsharma1911 i guess.

File: ui/components/Modal.js (hunk 1 above)
Comment (sudhanshutech) [reply ui/components/Modal.js:356]: we were till now using only i guess primary and secondary actions in these rjsf modals that is either we confirm or cancel

File: ui/components/MesheryMeshInterface/PatternService/RJSFCustomComponents/ObjectFieldTemplate.js [hunk 3]
```diff
@@ -13,11 +13,11 @@ import { ERROR_COLOR } from '../../../../constants/colors';
 import { iconMedium, iconSmall } from '../../../../css/icons.styles';
 import { calculateGrid } from '../helper';
 
-const styles = (theme) => ({
+const styles = () => ({
   objectFieldGrid: {
     padding: '.5rem',
     paddingTop: '0.7rem',
-    backgroundColor: theme.palette.type === 'dark' ? '#363636' : '#ffffff',
+    // backgroundColor: theme.palette.type === 'dark' ? '#363636' : '#ffffff',
```
Comment (Yashsharma1911) [inline ui/components/MesheryMeshInterface/PatternService/RJSFCustomComponents/ObjectFieldTemplate.js:20]: You have tested this Meshmap right :)

File: ui/components/Modal.js (hunk 1 above)
Comment (aabidsofi19) [reply ui/components/Modal.js:356]: @sudhanshutech  was this feedback acted upon ?

File: ui/components/MesheryMeshInterface/PatternService/RJSFCustomComponents/ObjectFieldTemplate.js (hunk 3 above)
Comment (sudhanshutech) [reply ui/components/MesheryMeshInterface/PatternService/RJSFCustomComponents/ObjectFieldTemplate.js:20]: yes , you can test  a open pr in extensions which has this following change

File: ui/components/Modal.js (hunk 1 above)
Comment (sudhanshutech) [reply ui/components/Modal.js:356]: for rjsf yes the rjsf wrapper is extracted and for the buttons slight modification needed so i have a upcoming pr shortly.

**Pull request merged**